components/brave_shields/core/common/brave_shields_settings_values.h

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_CORE_COMMON_BRAVE_SHIELDS_SETTINGS_VALUES_H_
+#define BRAVE_COMPONENTS_BRAVE_SHIELDS_CORE_COMMON_BRAVE_SHIELDS_SETTINGS_VALUES_H_
+
+#include <optional>
+
+#include "base/types/cxx23_to_underlying.h"
+#include "base/values.h"
+#include "brave/components/brave_shields/core/common/brave_shield_constants.h"
+#include "components/content_settings/core/common/content_settings_types.h"
+
+namespace brave_shields {
+
+enum ControlType {
+  ALLOW = 0,
+  BLOCK,
+  BLOCK_THIRD_PARTY,
+  DEFAULT,
+};
+
+namespace traits {
```

Comment:
This all feels a bit overly complicated for this task, wouldn't it be simpler to just have `ControlTypeToValue` and `ControlTypeFromValue` without all these templates?

Reply:
if we're going to do this, it should be in broken out into a separate PR with appropriate tests for all cases and then rebase this PR on top of it

Reply:
approved